native-schema-registry/csharp/AWSGsrSerDe/AWSGsrSerDe/serializer/avro/AvroSerializer.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Avro;
+using Avro.Generic;
+using Avro.IO;
+using Avro.Specific;
+using AWSGsrSerDe.common;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace AWSGsrSerDe.serializer.avro
+{
+    /// <summary>
+    /// Avro serialization helper.
+    /// </summary>
+    public class AvroSerializer : IDataFormatSerializer
+    {
+
+        private readonly MemoryCache _datumWriterCache;
+
+        /// <summary>
+        /// Size of datum writer cache for testing purpose
+        /// </summary>
+        public int CacheSize => _datumWriterCache.Count;
+
+        private AvroSerializer()
+        {
+            // TODO: make cache size limit configurable
+            _datumWriterCache = new MemoryCache(new MemoryCacheOptions{SizeLimit = 1000});
+        }
+
+        /// <summary>
+        ///  Constructor for Avro serializer that accepts configuration elements.
+        /// </summary>
+        /// <param name="configuration">configuration elements</param>
+        public AvroSerializer(GlueSchemaRegistryConfiguration configuration = null)
+        :this()
+        {
+        }
+
+        /// <summary>
+        /// Serialize the Avro message to bytes
+        /// </summary>
+        /// <param name="data">the Avro message for serialization</param>
+        /// <returns>the serialized byte array</returns>
+        public byte[] Serialize(object data)
+        {
+            // ReSharper disable once ConvertIfStatementToReturnStatement
+            if (data == null)
```

Comment:
Minor: Preferable to use multi-line if else statements here.

Reply:
`else` will be redundant in this case right?

Reply:
I meant,
```
if (data == null {
    return null
}
return Serialize...
``